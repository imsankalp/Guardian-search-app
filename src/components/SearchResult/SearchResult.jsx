import React from "react";
import PropTypes from "prop-types";
const SearchResult = ({ articles = [] }) => {
	return (
		<ul>
			{articles.map(({ webUrl, webTitle }) => (
				<li key={webUrl}>
					<a href={webUrl} target="_blank" rel="noopener noreferrer">
						{webTitle}
					</a>
				</li>
			))}
		</ul>
	);
};
SearchResult.propTypes = {
	articles: PropTypes.array,
};

export default SearchResult;
