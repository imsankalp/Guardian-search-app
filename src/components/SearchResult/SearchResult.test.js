import React from "react";
import Enzyme, { shallow } from "enzyme";
import SearchResult from "./SearchResult";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

const mockArticle = [
	{
		webUrl: "https://google.com",
		webTitle: "Google",
	},
];

describe("searchResult component", () => {
	test("renders", () => {
		const wrapper = shallow(<SearchResult articles={mockArticle} />);

		expect(wrapper).toMatchSnapshot();
	});

	test("return default empty array when no data to map through", () => {
		const wrapper = shallow(<SearchResult />);

		expect(wrapper).toMatchSnapshot();
	});

	test("Doesn't break when no article is passed", () => {
		const wrapper = shallow(<SearchResult />);

		expect(wrapper.find("li")).toHaveLength(0);
	});

	test("Doesn't break with empty array", () => {
		const wrapper = shallow(<SearchResult articles={[]} />);

		expect(wrapper.find("li")).toHaveLength(0);
	});
});
