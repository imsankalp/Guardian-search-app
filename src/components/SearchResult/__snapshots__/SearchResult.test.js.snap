// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchResult component renders 1`] = `ShallowWrapper {}`;

exports[`searchResult component return default empty array when no data to map through 1`] = `ShallowWrapper {}`;
